File: archive/aws-intro/go/lab-05/code/step1.go
package main

import (
	"encoding/json"

	"github.com/pulumi/pulumi-kubernetes/sdk/go/kubernetes/providers"
	"github.com/pulumi/pulumi/sdk/go/pulumi"
	"github.com/pulumi/pulumi/sdk/go/pulumi/config"
)

func main() {
	pulumi.Run(func(ctx *pulumi.Context) error {
		c := config.New(ctx, "")
		stackRef := c.Require("clusterStackRef")
		infra, err := pulumi.NewStackReference(ctx, stackRef, nil)

		kubeconfig := infra.GetOutput(pulumi.String("kubeconfig")).ApplyString(
			func(in interface{}) string {
				kc, err := json.Marshal(in)
				if err != nil {
					panic(err)
				}
				return string(kc)
			},
		)
		k8sProvider, err := providers.NewProvider(ctx, "k8sprovider", &providers.ProviderArgs{
			Kubeconfig: kubeconfig,
		})
		if err != nil {
			return err
		}

		return nil
	})
}

File: archive/aws-intro/python/lab-05/code/step1.py
from pulumi import export, StackReference, Output, ResourceOptions
from pulumi_kubernetes import Provider
import pulumi

# Create StackReference to the Kubernetes cluster stack
config  = pulumi.Config()
stackRef = config.require("clusterStackRef");
infra = StackReference(f"{stackRef}")

# Declare a provider using the KubeConfig we created
# This will be used to interact with the EKS cluster
k8s_provider = Provider("k8s-provider", kubeconfig=infra.get_output("kubeconfig"))
